roofit/roofitcore/inc/RooFuncWrapper.h

```diff
@@ -25,24 +26,27 @@
 #include <string>
 
 /// @brief  A wrapper class to store a C++ function of type 'double (*)(double* )'.
-/// The parameters can be accessed as x[<relative position of param in paramSet>] in the function body.
+/// The parameters can be accessed as params[<relative position of param in paramSet>] in the function body.
+/// The observables can be accessed as obs[i * num_entries + j], where i represents the observable position and j
+/// represents the data entry.
 /// @tparam Func Function pointer to the generated function.
-template <typename Func = double (*)(double *), typename Grad = void (*)(double *, double *)>
+template <typename Func = double (*)(double *, double *), typename Grad = void (*)(double *, double *, double *)>
 class RooFuncWrapper final : public RooAbsReal {
 public:
-   RooFuncWrapper(const char *name, const char *title, std::string const &funcBody, RooArgSet const &paramSet)
+   RooFuncWrapper(const char *name, const char *title, std::string const &funcBody, RooArgSet const &paramSet,
+                  RooArgSet const &ObsSet, RooAbsData &data)
       : RooAbsReal{name, title}, _params{"!params", "List of parameters", this}
    {
       std::string funcName = name;
-      std::string gradName = funcName + "_grad";
+      std::string gradName = funcName + "_grad_0";
       std::string requestName = funcName + "_req";
       std::string wrapperName = funcName + "_derivativeWrapper";
 
       gInterpreter->Declare("#pragma cling optimize(2)");
```

Comment:
Unrelated to this PR, but something that should be addressed before:

Can you undo the `gInterpreter->Declare("#pragma cling optimize(2)");` after the jitting is done? We don't want to change the global optimization level, which is happening here, right?

Reply:
Yes, okay, I think the default should be 0 then?

Reply:
Optimally you would query cling what the current optimization level is, is that possible?

The default depends on where you run ROOT even, for example it's 0 by default by 2 if you run on a cluster.

Probably if you ask on Mattermost shush how to temporarily change the cling optimization level we can solve this easily

Reply:
This is what I got from mattermost:

> yes, the simplest way is via an env variable: env EXTRA_CLING_ARGS='-O2' ./your_program

Maybe this is better as it allows you to run with other optimization levels also (for whatever reason that might be necessary...)

Reply:
But that's changing it for everything, not only for the JIT in RooFit :( Would be cool to find a way to do that only locally for the JITing of the function passed to the `RooFuncWrapper`.

So for now, can we leave out maybe the `gInterpreter->Declare("#pragma cling optimize(2)");` as I worry about the side effects, and for benchmarking we just do `EXTRA_CLING_ARGS='-O2'` as suggested?.

Reply:
Okay yes, sure.